apps/examples/lwnl_sample/lwnl_sample.c

```diff
@@ -0,0 +1,138 @@
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <net/if.h>
+#include <errno.h>
+#include <sys/select.h>
+#include <sys/socket.h>
+#include <tinyara/lwnl/lwnl.h>
+/*
+ * it's test code so it doesn't strictly error handling.
+ */
+static void *event_listener(void *arg)
+{
+	int res = 0;
+	int idx = *((int *)arg);
+	printf("[%d] run event listener\n", idx);
+	int fd = socket(AF_LWNL, SOCK_RAW, LWNL_ROUTE);
+	if (fd < 0) {
+		return NULL;
+	}
+
+	res = bind(fd, NULL, 0);
+	if (res < 0) {
+		printf("[%d] bind fail errno(%d)\n", idx, errno);
+		return NULL;
+	}
+
+	fd_set rfds, ofds;
+	FD_ZERO(&ofds);
+	FD_SET(fd, &ofds);
+
+	printf("[%d] run event loop\n", idx);
+	for (;;) {
+		rfds = ofds;
+		res = select(fd + 1, &rfds, NULL, NULL, 0);
+		if (res <= 0) {
+			printf("[%d] select error res(%d) errno(%d))\n", idx, res, errno);
+			return NULL;
+		}
+
+		if (FD_ISSET(fd, &rfds)) {
+			char buf[8] = {0, };
+			lwnl_cb_status status;
+			uint32_t len;
+			int nbytes = read(fd, (char *)buf, 8);
+			if (nbytes < 0) {
+				printf("[%d] read error bytes(%d) errno(%d))\n", idx, nbytes, errno);
+				return NULL;
+			}
+			memcpy(&status, buf, sizeof(lwnl_cb_status));
+			memcpy(&len, buf + sizeof(lwnl_cb_status), sizeof(uint32_t));
+			printf("[%d] status (%d) len(%d)\n", idx, status, len);
+
+			// flush the event queue
+			if (len > 0) {
+				char *tmp = (char *)malloc(len);
+				read(fd, tmp, len);
+				free(tmp);
+			}
+		}
+	}
+}
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int lwnl_sample_main(int argc, char *argv[])
+#endif
+{
+	pthread_t pid[3];
+
+	/*  run event listener thread */
+	for (int i = 0; i < 3; i++) {
```

Comment:
Can this test app have the number of test threads from a tester as an input parameter, not a fixed number ?
In this case, you should also add the help statement if the input parameter is missing.